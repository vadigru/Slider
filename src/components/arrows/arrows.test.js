import React from "react";
import renderer from "react-test-renderer";

import Arrows from "./arrows.jsx";

const indicators = [
  0,
  1,
  2,
];

it(`Should render Arrows component`, () => {
  const tree = renderer
    .create(
        <Arrows
          activeSlide={2}
          isInfinite={true}
          isDisabled={false}
          isDragging={false}
          slidesToShow={2}
          indicators={indicators}
          onLeftArrowClick={() => {}}
          onRightArrowClick={() => {}}
        />,
        {
          createNodeMock: () => {
            return document.createElement(`div`);
          }
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
