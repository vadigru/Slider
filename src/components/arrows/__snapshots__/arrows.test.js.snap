// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Arrows component 1`] = `
<div
  className="arrows"
>
  <button
    className="arrows-btn arrow-prev
          
          "
    disabled={false}
    id="arrow-prev"
    onClick={[Function]}
  >
    <svg
      className="arrow-prev-icon"
      viewBox="0 0 20 20"
    >
      <path
        d="M8.388,10.049l4.76-4.873c0.303-0.31,0.297-0.804-0.012-1.105c-0.309-0.304-0.803-0.293-1.105,0.012L6.726,9.516c-0.303,0.31-0.296,0.805,0.012,1.105l5.433,5.307c0.152,0.148,0.35,0.223,0.547,0.223c0.203,0,0.406-0.08,0.559-0.236c0.303-0.309,0.295-0.803-0.012-1.104L8.388,10.049z"
        fill="#cccccc"
      />
    </svg>
  </button>
  <button
    className="arrows-btn arrow-next
          
          "
    disabled={false}
    id="arrow-next"
    onClick={[Function]}
  >
    <svg
      className="arrow-next-icon"
      viewBox="0 0 20 20"
    >
      <path
        d="M11.611,10.049l-4.76-4.873c-0.303-0.31-0.297-0.804,0.012-1.105c0.309-0.304,0.803-0.293,1.105,0.012l5.306,5.433c0.304,0.31,0.296,0.805-0.012,1.105L7.83,15.928c-0.152,0.148-0.35,0.223-0.547,0.223c-0.203,0-0.406-0.08-0.559-0.236c-0.303-0.309-0.295-0.803,0.012-1.104L11.611,10.049z"
        fill="#cccccc"
      />
    </svg>
  </button>
</div>
`;
